nemo/deploy/deploy_base.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+from abc import ABC, abstractmethod
+
+try:
+    from pytorch_lightning import Trainer
+except Exception:
+    pass
+
+from nemo.deploy.triton_deployable import ITritonDeployable
+
+use_nemo = True
+try:
+    from nemo.core.classes.modelPT import ModelPT
+except:
+    use_nemo = False
+
+
+class DeployBase(ABC):
+    def __init__(
+        self,
+        triton_model_name: str,
```

Comment:
Another thing - if this is going to be part of NeMo core, please do not specialize to LLM domain as the only option. NeMo's core is meant to be general to all domains underneath its scope (its domains are separated only at the `colelctions` level).

This abstract class should have minimum domain specific design. You can have a subclass of this that specialized to LLM models, and subclass that specializes to TRT based LLM or VLLM based LLM deployment (as an example for the future, not that you need to actually develop vllm based deployment in this PR).

Reply:
Not sure what is specific to LLM here. Seems to me generic enough but if you think any parameter shouldn't be here, happy to remove it.

Reply:
An example would be the activation checkpointing logic wrapped in try catch. In the future, there can potentially be more such cases. 

This is a base class, it shouldn't specialize on domain, domain specific subclasses would be better in my opinion so that such logic can be located in only the domains that would need them.

Reply:
The only code in the base class is the _init_nemo_model function which should be generic to any ckpt in the nemo. It's using the restore_from function in the modelPT class which I think is not specific to LLM. And it gets the class info from the generic config file and converts the object to that class which is still not specific to LLM.

I really designed this in a way that any model in nemo can use it.

Reply:
Alright if need be it can be refactored in the future anyway